app/gui/src/model/module/synchronized.rs

```diff
@@ -426,36 +416,37 @@ impl Module {
                     };
                     //id_map goes first, because code change may alter its position.
                     let edits = vec![id_map_change, code_change];
-                    let summary = &summary.summary;
-                    let notify_ls = self.notify_language_server(summary, &new_file, edits, true);
+                    let summary_digest = summary.summary.digest;
+                    let notify_ls =
+                        self.notify_language_server(summary_digest, &new_file, edits, true);
                     profiler::await_!(notify_ls, _profiler)
                 }
                 NotificationKind::MetadataChanged => {
                     let edits = vec![TextEdit {
                         range: summary.metadata_engine_range().into(),
                         text:  new_file.metadata_slice().to_string(),
                     }];
-                    let summary = &summary.summary;
-                    let notify_ls = self.notify_language_server(summary, &new_file, edits, false);
+                    let summary_digest = summary.summary.digest;
+                    let notify_ls =
+                        self.notify_language_server(summary_digest, &new_file, edits, false);
                     profiler::await_!(notify_ls, _profiler)
                 }
-                NotificationKind::Reloaded => Ok(ParsedContentSummary::from_source(&new_file)),
+                NotificationKind::Reloaded => {}
             },
         }
     }
 
-    /// Send update to Language Server with the entire file content. Returns the new content summary
-    /// of Language Server state.
+    /// Send update to Language Server with the entire file content.
     #[profile(Debug)]
     fn full_invalidation(
         &self,
-        ls_content: &ContentSummary,
+        ls_content: ContentSummary,
         new_file: SourceFile,
-    ) -> impl Future<Output = FallibleResult<ParsedContentSummary>> + 'static {
+    ) -> impl Future<Output = ()> + 'static {
```

Comment:
Does the `Output` not default to `()` ? If not, this is an IMO surprising design of this lib.

Reply:
Associated type defaults are not in stable Rust yet and unstable feature is incomplete, especially around trait objects.
https://github.com/rust-lang/rust/issues/29661 ...since 2015!